networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/encodings/compression/snappy/SnappyFrameDecoder.java

```diff
@@ -194,14 +195,20 @@ public SnappyFrameDecoder(boolean validateChecksums) {
           snappy.reset();
           break;
       }
-    } catch (Exception e) {
+    } catch (CompressionException e) {
       corrupted = true;
       throw e;
     }
     return Optional.ofNullable(ret);
   }
 
-  private static void checkByte(byte actual, byte expect) {
+  @Override
+  protected void throwDataTruncatedException(int dataLeft) throws CompressionException {
+    throw new PayloadSmallerThanExpectedException(
```

Comment:
I think this should throw the opposite error - we think we're done processing the response but there's extra data in the pipeline, so the payload was larger than we expected:
```suggestion
    throw new PayloadLargerThanExpectedException(
```

Reply:
No, here it means that we were requesting the next message to be decoded, there was just a part of this message which was stored and then the stream _prematurely_ completed. So the last message appeared _truncated_